accepted/2025/cli-acquisition-tool.md

```diff
@@ -0,0 +1,223 @@
+
+## Q & A
+
+### How will I get `dnup`?
+
+We intend to distribute `dnup` via a direct download initially.
+The dotnet download web site and the Learn documentation for getting started with .NET are the primary candidates.
+However, it is also common to distribute these kinds of bootstrappers via system package managers, so future efforts could include WinGet packages, Debian and Red Hat packages, a Homebrew cask, etc.
+The tool itself will be a Native AOT tool that is completely standalone, so acquisition should be relatively simple to set up.
+
+### How will this handle existing global installations?
+
+`dnup` will detect existing global installations but focus on managing user-local installs.
+Users with existing package manager installations will continue to use those tools. `dnup` will provide guidance when detecting unsupported installation types, and may offer to seed the user-local hive with the same versions so that there are no functionality gaps when starting with `dnup`.
+
+### What about impact on services running under different accounts?
+
+For users with services running under different accounts using `dotnet.exe`, setting user-specific PATH and DOTNET_ROOT may cause those services to fail.
+We expect this to be a corner case since launching services that way is not recommended, but we need to document this publicly and gather feedback.
+
+### How does this integrate safely with IDEs?
+
+IDEs like Visual Studio Code often have a "bring your own" approach to SDKs and tooling.
+This tool works hand in hand with that model - enabling users to more easily acquire whatever version of the SDK they need to work with.
+
+Visual Studio has an execution model that is more tightly bound to the SDK.
+While the .NET Runtime, MSBuild, and SDK teams have begun work to enable less coupling here (via the global.json `sdk.paths` feature, the `DOTNET_HOST_PATH` AppHost Switch, and MSBuild SDKResolver enhancements for .NET 10), there is still some coupling that could result in negatively impacting the end-user experience.
+
+To help get ahead of this, we have begun work with the VS team to expand the set of cross-SDK/cross-VS version testing that is done in VS today.
+We will aim to keep the SDK and VS maximally-compatible so that VS can support consuming the SDKs across a broader range than previously documented.
+
+It's worth noting that this need for compatibility is not a problem unique to `dnup` - it exists today when users use locally-managed SDKs installed by Arcade, dotnet-install scripts, or other means.
+_Where_ the SDK is resolved from is largely orthogonal to _how compatible_ that SDK is with the IDE.
+
+### What future work does this enable?
+
+- Unifying IDE-specific and disparate install components that exist today
+  - Specifically components like the `dotnet-install` scripts and all downstream consumers of it (`actions/setup-dotnet`, `UseDotNet`)
```

Comment:
Will the tool have flags to enable replacing the scripts to acquire nightly builds too, not just the official releases?

Reply:
Nightlies are a much harder thing but definitely something I am interested in as the tool grows. We will need to work with the VMR to create some parallels for the data we get from the releases.json manifests, but fundamentally I'd like this tool to eventually support
* stable releases
* nightlies from the VMR
* PRs from VMR (or SDK?)

And depending on how the long-term goal of 'factoring out the SDK into pay-as-you-go chunks' goes, you could potentially see grabbing PRs/nightlies of `specific_component_X` in the far future.